packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/InboundLineEdit.tsx

```diff
@@ -14,130 +14,266 @@ import {
-  useEffect(() => {
-    if (
-      wrappedInbound?.batches &&
-      Object.values(wrappedInbound?.batches).length === 0
-    ) {
-      onAddBatch();
-    }
-  });
-
-  return loading ? (
-    <BasicSpinner />
-  ) : (
-    <>
-      <InboundLineEditForm
-        draft={draft}
-        mode={mode}
-        item={item}
-        onChangeItem={onChangeItem}
-      />
-      <Divider margin={5} />
-      {wrappedInbound ? (
-        <TabContext value={currentTab}>
-          <Box flex={1} display="flex" justifyContent="space-between">
-            <Box flex={1} />
-            <Box flex={1}>
-              <TabList
-                value={currentTab}
-                centered
-                onChange={(_, v) => setCurrentTab(v)}
-              >
-                <Tab value={Tabs.Batch} label={Tabs.Batch} />
-                <Tab value={Tabs.Pricing} label={Tabs.Pricing} />
-              </TabList>
-            </Box>
-            <Box flex={1} justifyContent="flex-end" display="flex">
-              <ButtonWithIcon
-                color="primary"
-                variant="outlined"
-                onClick={onAddBatch}
-                label={t('label.add-batch')}
-                Icon={<PlusCircleIcon />}
-              />
-            </Box>
-          </Box>
-
-          <TableContainer
-            sx={{
-              height: isMediumScreen ? 300 : 400,
-              marginTop: 2,
-              borderWidth: 1,
-              borderStyle: 'solid',
-              borderColor: 'divider',
-              borderRadius: '20px',
-            }}
-          >
-            <Box
-              sx={{
-                width: 400,
-                height: isMediumScreen ? 300 : 400,
-                backgroundColor: theme =>
-                  alpha(theme.palette['background']['menu'], 0.4),
-                position: 'absolute',
-                borderRadius: '20px',
-              }}
-            />
-            <StyledTabPanel value={Tabs.Batch}>
-              <BatchTable batches={batches} />
-            </StyledTabPanel>
-
-            <StyledTabPanel value={Tabs.Pricing}>
-              <PricingTable batches={batches} />
-            </StyledTabPanel>
-          </TableContainer>
-        </TabContext>
+  return (
+    <Modal
+      title={
+        mode === ModalMode.Create
+          ? t('heading.add-item')
+          : t('heading.edit-item')
+      }
+      cancelButton={<DialogButton variant="cancel" onClick={onClose} />}
+      nextButton={
+        <DialogButton
+          variant="next"
+          onClick={async () => {
+            try {
+              await saveLines();
+              setCurrentItem(mode === ModalMode.Update ? nextItem : null);
+              return true;
+            } catch (e) {
+              return false;
+            }
+          }}
+        />
+      }
+      okButton={<DialogButton variant="ok" onClick={saveLines} />}
+      height={600}
+      width={1024}
+    >
+      {isLoading ? (
+        <BasicSpinner />
```

Comment:
still bothers me that the spinner says 'loading' when it is saving. what about a 'mode' prop on the spinner to indicate the action you are waiting on? In this modal we are only waiting for save, not load, so that should be easy enough to use

Reply:
Yes, sorry not forgotten (maybe a little!!). Will do!